### PR TITLE
Add basic docs for dynamic forall

### PR DESCRIPTION
**Pull request by artv3**

### CODE REVIEW

**Comment by artv3** [comment]
I ran the white-space remover in my editor and made more changes than expected. To simplify review I suggest hiding white-space changes.

File: docs/sphinx/user_guide/feature/loop_basic.rst [hunk 1]
```diff
@@ -12,21 +12,21 @@
...
 ==============================================
 
-The ``RAJA::forall``, ``RAJA::kernel``, and ``RAJA::expt::launch`` 
+The ``RAJA::forall``, ``RAJA::expt::dynamic_forall``, ``RAJA::kernel``, and ``RAJA::expt::launch``
 template methods comprise the RAJA interface for kernel
-execution. ``RAJA::forall`` methods execute simple, non-nested loops, 
-``RAJA::kernel`` methods support nested loops and other complex loop 
-kernels and transformations, and ``RAJA::expt::launch`` creates an execution 
-space in which kernels are written in terms of nested loops using 
+execution. ``RAJA::forall`` methods execute simple, non-nested loops,
```
Comment (rhornung67) [inline docs/sphinx/user_guide/feature/loop_basic.rst:17]:
```suggestion
execution. ``forall`` methods execute simple, non-nested loops,
```

File: docs/sphinx/user_guide/feature/loop_basic.rst [hunk 2]
```diff
@@ -12,21 +12,21 @@
...
+space in which kernels are written in terms of nested loops using
 the ``RAJA::expt::loop`` method.
 
-.. note:: The ``forall`` , and ``kernel`` methods are in the ``RAJA`` 
-          namespace, while ``launch`` is in the RAJA namespace for 
-          experimental features ``RAJA::expt``.  ``RAJA::expt::launch`` 
+.. note:: The ``forall`` , and ``kernel`` methods are in the ``RAJA``
+          namespace, while ``dynamic_forall`` and ``launch`` are in the RAJA namespace for
+          experimental features ``RAJA::expt``.  ``RAJA::expt::launch``
           will be moved to the ``RAJA`` namespace in a future RAJA release.
```
Comment (rhornung67) [inline docs/sphinx/user_guide/feature/loop_basic.rst:26]:
```suggestion
          and ``RAJA::expt::dynamic_forall`` will be moved to the ``RAJA`` namespace in a future RAJA release.
```

File: docs/sphinx/user_guide/feature/loop_basic.rst [hunk 3]
```diff
@@ -53,28 +53,42 @@ This may be written using ``RAJA::forall`` as::
...
+				    ...
+				   pol_N>;
+  int pol = i; //run-time value
+
+  RAJA::expt::dynamic_forall<exec_pol_list>(pol, RAJA::TypedRangeSegment<int>(0, N), [=] (int i) {
+     c[i]  = a[i] + b[i];
+  });
+
+
+While loop execution using ``RAJA::forall`` methods is a subset of
```
Comment (rhornung67) [inline docs/sphinx/user_guide/feature/loop_basic.rst:83]: Confirming that this statement **does not** apply to RAJA::expt::dynamic_forall.

**Review by rhornung67 — COMMENTED** [review]
A few minor comment that I think are worth addressing.

File: docs/sphinx/user_guide/feature/loop_basic.rst (hunk 3 above)
Comment (artv3) [reply docs/sphinx/user_guide/feature/loop_basic.rst:83]: I think we can include ``RAJA::dynamic_forall`` in the paragraph, we would just have to note that kernel does not support dynamic policy selection.

File: docs/sphinx/user_guide/feature/loop_basic.rst [hunk 4]
```diff
@@ -53,28 +53,42 @@ This may be written using ``RAJA::forall`` as::
...
+				    ...
+				   pol_N>;
+  int pol = i; //run-time value
+
+  RAJA::expt::dynamic_forall<exec_pol_list>(pol, RAJA::TypedRangeSegment<int>(0, N), [=] (int i) {
+     c[i]  = a[i] + b[i];
+  });
+
+
+While static loop execution using ``forall`` methods is a subset of
```
Comment (artv3) [inline docs/sphinx/user_guide/feature/loop_basic.rst:83]: @rhornung67  would the change above work?

File: docs/sphinx/user_guide/feature/loop_basic.rst (hunk 3 above)
Comment (rhornung67) [reply docs/sphinx/user_guide/feature/loop_basic.rst]: works for me.

File: docs/sphinx/user_guide/feature/loop_basic.rst (hunk 4 above)
Comment (rhornung67) [reply docs/sphinx/user_guide/feature/loop_basic.rst:83]: I think so.

**Review by rhornung67 — APPROVED** [review]

**Comment by rhornung67** [comment]
@artv3 did you force this merge b/c the CI was timing out?

**Comment by artv3** [comment]
> @artv3 did you force this merge b/c the CI was timing out?
> 
> If you no longer need the branch, please delete it.

Hi @rhornung67 , I used the auto-merge option. It merges the branch when all the required checks pass. If it merged when it was not supposed to I can revert the change.

**Comment by rhornung67** [comment]
@artv3 I saw you had auto-merge enabled, which is fine. I was confused because it was merged and it looked like there were some CI failures.

**Pull request merged**